[Bug] umi-scripts: postinstall会在路径含有空格时报错

<!--
感谢您向我们反馈问题，为了高效的解决问题，我们期望你能提供以下信息：
-->

## What happens?
umi-scripts: postinstall

Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/Users/youniverse/Personal' imported from /Users/youniverse/Personal Code/umi/
<!-- A clear and concise description of what the bug is. -->
<!-- 清晰的描述下遇到的问题。-->

## Mini Showcase Repository(REQUIRED) 

与我的fork仓库无关，是与本地路径相关的问题。
...这机器人是一定要我提供一个仓库地址吗
https://github.com/FriedRiceNoodles/umi

> Please provide a [minimal reproduction](https://stackoverflow.com/help/minimal-reproducible-example) then upload to your GitHub. 请提供 [最小重现](https://stackoverflow.com/help/minimal-reproducible-example)，并上传到你的 GitHub 仓库

<!-- 为节约大家的时间，无复现步骤的 ISSUE 会被关闭，提供之后再 REOPEN -->
<!-- YOUR_REPOSITORY_URL on github or stackbliz -->

## How To Reproduce
我的本地路径是：/Users/youniverse/Personal Code/umi/
报错：Cannot find module '/Users/youniverse/Personal' imported from /Users/youniverse/Personal Code/umi/

<!-- 请提供复现链接/步骤，错误日志以及相关配置 -->

## Context

- **Umi Version**: 4.0.33
- **Node Version**: 16.15.0
- **Platform**: MacOS

--- a/scripts/bin/umi-scripts.js
+++ b/scripts/bin/umi-scripts.js
@@ -25,9 +25,12 @@ if (throughArgs.length) {
   throughArgs.unshift('--')
 }
 
+// current dir path may contain spaces
+// https://github.com/umijs/umi/issues/9865
+const scriptPathAsStr = JSON.stringify(scriptsPath)
 const spawn = sync(
   'tsx',
-  [scriptsPath, ...throughArgs],
+  [scriptPathAsStr, ...throughArgs],
   {
     env: process.env,
     cwd: process.cwd(),

